hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieTestDataGenerator.java

```diff
@@ -607,8 +608,14 @@ public List<HoodieRecord> generateInsertsForPartition(String instantTime, Intege
   }
 
   public Stream<HoodieRecord> generateInsertsStream(String commitTime, Integer n, boolean isFlattened, String schemaStr, boolean containsAllPartitions) {
+    AtomicInteger partitionIndex = new AtomicInteger(0);
     return generateInsertsStream(commitTime, n, isFlattened, schemaStr, containsAllPartitions,
-        () -> partitionPaths[rand.nextInt(partitionPaths.length)],
+        () -> {
+          // round robin to ensure we generate inserts for all partition paths
+          String partitionToUse = partitionPaths[partitionIndex.get()];
+          partitionIndex.set((partitionIndex.get() + 1) % partitionPaths.length);
+          return partitionToUse;
```

Comment:
so with this change we can guarantee all target partitions have records. then we don't need to bump 10 records to 100? so we can make it faster. We just need to make sure num records > num partitions here

Reply:
yes, but I don't expect any savings from 100 to 10 records. Also most of our tests are doing 100 recs. so just to keep it in parity.

Reply:
this is minor: 1) from UT perspective, 100 records shouldn't be different from 10 records. so why not keep it less. 2) we should ensure num records > num partitions without knowing there are always 3 partitions so some check arg will help prevent misuse leading to unexpected results